tests/future/test_future.cmake

```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright 2021, Intel Corporation
+
+# an example for the basic test case
+
+include(${SRC_DIR}/cmake/test_helpers.cmake)
+
+setup()
+
+# The expected input can be provided to the execute function,
+# which will check if the return code of the binary file matches.
+execute(0 ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${BUILD}/dummy)
```

Comment:
I think you are accidentally testing a wrong binary.

Reply:
So that's why it worked on the first try :D Good catch.
Fixed.